Upgrade jscodeshift to avoid using the broken colors library

zhanwang626 opened the pull request:
https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/ 

Since the `colors@1.4.1` was broken, `jscodeshift` decided to use `chalk` instead of `colors`. https://github.com/facebook/jscodeshift/pull/473

Let's upgrade the `jscodeshift` to the latest version to avoid using the broken `colors` library.

ckundo commented on yarn.lock:
```diff
@@ -1834,7 +2090,7 @@ color-name@~1.1.4:
   integrity sha512-dOy+3AuW3a2wNbZHIuMZpTcgjGuLU/uBL/ubcZF9OXbDo8ff4O8yVp5Bf0efS8uEoYo5q4Fx7dY9OgQGXgAsQA==
 
-colors@^1.1.2, colors@^1.4.0:
+colors@^1.4.0:
```
shouldn't this have gone away?

zhanwang626 replied:
> shouldn't this have gone away?

It should but unfortunately "ember-template-recast" depends on `colors`
```
yarn why v1.22.10
warning ../../package.json: No license field
[1/4] 🤔  Why do we have the module "colors"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "colors@1.4.0"
info Reasons this module exists
   - "ember-template-recast" depends on it
   - Hoisted from "ember-template-recast#colors"
info Disk size without dependencies: "116KB"
info Disk size with unique dependencies: "116KB"
info Disk size with transitive dependencies: "116KB"
info Number of shared dependencies: 0
✨  Done in 0.56s.
```

rwjblue commented:
Any breaking changes in the jscodeshift update that we need to worry about?

rwjblue commented:
@zhanwang626 - Mind rebasing so that we can get a CI run kicked off?

zhanwang626 commented:
> Any breaking changes in the jscodeshift update that we need to worry about?

It doesn't seem like there are any breaking changes in the jscodeshift:

- Switched from colors to chalk to mitigate a security vulnerability in colors@1.4.1
- Added a `--fail-on-error` flag to return a `1` error code when errors were found
- Created `template.asyncExpression`
- Removed lodash dependency from tsx parser
- Allow transform to be a Promise
- Support newer TypeScript syntax by upgrading to newer Babel parser

rwjblue commented:
Gotcha, still needs a rebase (or empty commit) to trigger a CI run...

zhanwang626 commented:
> Gotcha, still needs a rebase (or empty commit) to trigger a CI run...

Cool, will do a empty commit now.

scalvert commented:
Looks like CI failed due to an unresolved package.

zhanwang626 commented:
@rwjblue Due to the version of eslint upgrade, it seems the CI test engine's node 10 env is not compatible with this project anymore. https://github.com/rwjblue/codemod-cli/runs/4796351967?check_suite_focus=true#step:5:23
